src/Profile/DeviceConfig.cpp

```diff
@@ -100,29 +100,21 @@ LoadPath(const ProfileMap &map, DeviceConfig &config, unsigned n)
 {
   char buffer[64];
   MakeDeviceSettingName(buffer, "Port", n, "Path");
-  return map.Get(buffer, config.path);
-}
-
-static bool
-LoadPortIndex(const ProfileMap &map, DeviceConfig &config, unsigned n)
-{
-  char buffer[64];
-  MakeDeviceSettingName(buffer, "Port", n, "Index");
-
-  unsigned index;
-  if (!map.Get(buffer, index))
+#ifdef _WIN32 
+  // on Windows don't use the form 'COM1:' anymore, use 'COM1'
+  if (map.Get(buffer, config.path)) {
+    if (config.path[config.path.length() - 1] == ':') {
```

Comment:
Buffer overflow bug here; and a very clumsy way to access `config.path.back()`.

Reply:
clumsy - okay. I am and will remain a newbie in your framework... And I see the problem with length() == 0 now!

Reply:
That's not "my framework", that's the C++ standard library; are you a newbie to C++?
(See: https://en.cppreference.com/w/cpp/container/vector/back https://en.cppreference.com/w/cpp/container/array/back https://en.cppreference.com/w/cpp/container/list/back etc.)

Reply:
I never worked with std::array up to now, so I'm a newbie with this!